### PR TITLE
Allow for network role to be fully optional

### PR DESCRIPTION
**Pull request by jvonau**
Alternate to #3145

### CODE REVIEW

**Comment by jvonau** [comment]
discussion in #2876 - now opt out.

**Comment by holta** [comment]
What testing paths are likely most important for this PR?

Thanks everyone if you can help us out here, to confirm this work !

**Comment by jvonau** [comment]
https://github.com/iiab/iiab/issues/3106#issuecomment-1140557739

**Comment by holta** [comment]
This looks extremely promising :heavy_check_mark:

I spoke to @deldesir who is very busy in July but regardless I'm hoping several of us can test this under various conditions so it can be merged well before the end of July.

**Comment by jvonau** [comment]
The default install flow is not altered with the defaults being True, just rearranged. You have the hood up toss it in before I have to rebase again, getting tired of the circus games. I'll write something up to explain it once the code is place, not wasting my time otherwise.

File: roles/network/tasks/install.yml [hunk 1]
```diff
@@ -52,10 +55,32 @@
     - { name: 'net.ipv4.ip_forward', value: '1' }  # Masquerading LAN->Internet
     - { name: 'net.ipv4.conf.default.rp_filter', value: '1' }
     - { name: 'net.ipv4.conf.default.accept_source_route', value: '0' }
-    #- { name: 'kernel.sysrq', value: '1' }             # OS values differ, Ok?
-    - { name: 'kernel.core_uses_pid', value: '1' }
```
Comment (holta) [inline roles/network/tasks/install.yml:56]: Ansible output `'kernel.core_uses_pid: 1' +` should be removed ~8 lines above.

**Comment by holta** [comment]
> I'll write something up to explain it

:+1:

Articulating test strategies is something I'll help with in July.

I'm recruiting new people to help with testing, and they will need testing strategies/tactics outlined (very helpful for everyone!)

Regardless how [roles/network](https://github.com/iiab/iiab/tree/master/roles/network) and other things evolve over time!

File: roles/network/tasks/install.yml (hunk 1 above)
Comment (jvonau) [reply roles/network/tasks/install.yml:56]: You can fix it, I'm packing to leave for the weekend

Comment (jvonau) [reply roles/network/tasks/install.yml:56]: never mind I had a minute to spare

**Comment by holta** [comment]
Thanks to @avni who smoke-tested this PR yesterday: http://sprunge.us/IE01pP?en

```
root@box:~# iiab-summary
install_date = 2022-07-04T16:20:40Z   Current TZ: EDT

iiab: fc7a7419, 21 PR's / 83 commits since tag 8.0-preview-2
    "Merge branch 'wifi_net2' of https://github.com/jvonau/iiab into wifi_net2"
       branch: wifi_net2

/etc/iiab/pr-list-pulled:
3173 2022-07-04 12:19:38 EDT

iiab-admin-console: c6c6b9d, 43 PR's / 123 commits since tag v0.5.5.9
    "Merge pull request #505 from tim-moody/0.6.0.4-enh2"
    https://github.com/iiab/iiab-admin-console   branch: master

Ubuntu 22.04 LTS   bookworm/sid
display-manager? inactive   Arch1: amd64   Arch2:
5.15.0-40-generic #43-Ubuntu SMP Wed Jun 15 12:54:21 UTC 2022 x86_64
Intel(R) Core(TM) i5-8279U CPU @ 2.40GHz   CPU(s): 2   Mem: 1972 Swap: 0

Filesystem      Size  Used Avail Use% Mounted on
/dev/sda1        25G  3.7G   22G  15% /      ZIMs: 0 OER2Go: 0 Apps2B: 0

lo enp5s0 enp6s0 tun0
openvpn_enabled: True
openvpn_handle: 204-u22-srv-PR3173
10.161.33.179 192.168.1.204 10.8.0.26 fd42:5288:44a8:b337:5054:ff:fea5:b9e7
```

**Comment by jvonau** [comment]
> 9 iiab commit:               fc7a741965bfab017a7d2c70e4bd2dda1985834b

Think I would prefer to see the last git hash of the branch that is being tested here, but that is harder to pull off.

>  10      remote:               
  11      branch:               wifi_net2

Nice, something more meaningful. 

>  12   21 merged PR's since recent tag: 8.0-preview-2 (83 commits)
 113 iiab_remote = none
 124 iiab_remote = none

Thought I would see "https://github.com/iiab/iiab" at 10,113,124

> 642 =IIAB==========================================================================
 643 -rw-r--r-- 1 root root 367 Jul  4 12:24 /etc/iiab/iiab_state.yml
 644 
 645 # DO *NOT* MANUALLY EDIT THIS, THANKS!
 646 # IIAB does NOT currently support uninstalling apps/services.
 647 
 648 network_installed: True
 649 sshd_installed: True
 650 openvpn_installed: True
 651 remoteit_installed: True
 652 iiab_admin_installed: True
 653 mysql_installed: True
 654 nginx_installed: True
 655 www_base_installed: True
 656 pylibs_installed: True
 657 usb_lib_installed: True
 658 www_options_installed: True

Correct order of events.

**Comment by holta** [comment]
> harder to pull off.

Obtaining [info about the branch being tested] isn't terribly hard, e.g. on @avni's vm4 here:

```
root@box:/opt/iiab/iiab# git log HEAD^2 -1
commit 11f451da1af99b6eb8f47a808c9a6be8606fb381
Author: Jerry Vonau <jvonau3@gmail.com>
Date:   Thu Jun 30 12:21:33 2022 -0500

    visual feedback
```

```
root@box:/opt/iiab/iiab# git log HEAD^2 -1 --oneline
11f451da visual feedback
```

```
root@box:/opt/iiab/iiab# git log HEAD^2 -1 --pretty=format:%s
visual feedback
```

> > 10      remote:

> > 113 iiab_remote = none
> > 124 iiab_remote = none
> 
> Thought I would see "https://github.com/iiab/iiab" at 10,113,124

We should avoid an archeological dig snooping through volunteers' command-line histories.

But yes if absolutely necessary this is possible, e.g. obtainable from Git's metadata database: (which records when the tips of branches and other references were updated in the local repository)

```
root@box:/opt/iiab/iiab# git reflog | head -1
fc7a7419 HEAD@{0}: pull --no-edit --no-rebase https://github.com/jvonau/iiab wifi_net2: Merge made by the 'ort' strategy.
```

Another big issue is [iiab-summary](https://github.com/iiab/iiab/blob/master/scripts/iiab-summary) and [iiab-diagnostics](https://github.com/iiab/iiab/blob/master/scripts/iiab-diagnostics.README.md) already flood ppl with TMI = too much information = Three Mile Island ;)

Let me think about this.

I'll try to add 1-or-2 more such things before IIAB 8.0 is released hopefully in August.

**Comment by jvonau** [comment]
The last git hash of the branch is what is important, that is what ties what is in the branch to what is being tested.
>root@box:/opt/iiab/iiab# git log HEAD^2 -1 --oneline
11f451da visual feedback

I like this one for the reason above.

**Comment by jvonau** [comment]
#3298

**Comment by holta** [comment]
> This looks extremely promising :heavy_check_mark:
> 
> I spoke to @deldesir who is very busy in July but regardless I'm hoping several of us can test this under various conditions so it can be merged well before the end of July.

Done!  Please see:

- PR #3302

Thanks to @avni for helping to test/evaluate.

> I'll write something up to explain it once the code is place

This will help a lot of people!  Wherever it's placed:

- https://github.com/iiab/iiab/tree/master/roles/network#readme
- https://github.com/iiab/iiab/wiki/IIAB-Networking
- http://faq.iiab.io -> "23 Any other networking tips?"
- _Etc_

**Pull request closed without merging**